### PR TITLE
Add GUI prompts for GitLab authentication

### PR DESCRIPTION
**Pull request by mjcheetham**
Add GUI prompts on all platforms for GitLab authentication. GUI prompts are required because if GCM/Git is invoked from an IDE without a terminal, there is no way for the user to authenticate for GitLab.

Browser|Token|Password
-|-|-
image|image|image

Note that if the host is "gitlab.com" then the instance name is not shown.
The "Sign Up" link takes you to <https://about.gitlab.com>.

### CODE REVIEW

File: src/shared/GitLab.UI.Avalonia/Program.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+            // Run AppMain in a new thread and keep the main thread free
+            // to process the dispatcher's job queue.
+            var appMain = new Thread(AppMain) {Name = nameof(AppMain)};
+            appMain.Start(args);
+
+            // Process the dispatcher job queue (aka: message pump, run-loop, etc...)
+            // We must ensure to run this on the same thread that it was created on
+            // (the main thread) so we cannot use any async/await calls between
+            // Dispatcher.Create and Run.
```
Comment (ldennington) [inline src/shared/GitLab.UI.Avalonia/Program.cs:36]: Nit: should this be `Dispatcher.Initialize`?

File: src/shared/GitLab.UI/ViewModels/CredentialsViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+                case nameof(Token):
+                    SignInTokenCommand.RaiseCanExecuteChanged();
+                    break;
+            }
+        }
+
+        private void SignUp()
+        {
+            BrowserUtils.OpenDefaultBrowser(_environment, "https://about.gitlab.com/");
```
Comment (hickford) [inline src/shared/GitLab.UI/ViewModels/CredentialsViewModel.cs:63]: URL should depend on host, eg. https://gitlab.example.com/users/sign_up or simply https://gitlab.example.com/

**Review by hickford — CHANGES_REQUESTED** [review]

**Comment by hickford** [comment]
> Looks good. Do we want @hickford to give this a quick test?

Works for me.

**Review by ldennington — APPROVED** [review]
B-e-a-utiful!

**Review by hickford — CHANGES_REQUESTED** [review]

File: src/shared/GitLab.UI.Avalonia/Views/CredentialsView.axaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+             xmlns:vm="clr-namespace:GitLab.UI.ViewModels;assembly=GitLab.UI.Shared"
+             xmlns:converters="clr-namespace:GitCredentialManager.UI.Converters;assembly=gcmcoreuiavn"
+             mc:Ignorable="d" d:DesignWidth="420"
+             x:Class="GitLab.UI.Views.CredentialsView">
+    <Design.DataContext>
+        <vm:CredentialsViewModel/>
+    </Design.DataContext>
+    <DockPanel>
+        <StackPanel DockPanel.Dock="Top" Margin="0,0,0,15">
+            <Image Source="../Assets/gitlab-logo-gray-rgb.png"
```
[the remark this answers is not in the document]
Comment (mjcheetham) [reply src/shared/GitLab.UI.Avalonia/Views/CredentialsView.axaml]: I've updated the PR to now use a SVG-based rendering of the "fox" logo, and the logotype "GitLab", separately.

We now hide the logo when it's not gitlab.com.

**Comment by mjcheetham** [comment]
New screenshots with SVG-based logo and logotype, as well as for gitlab.com and custom instances (for both Avalonia and WPF implementations).

gitlab.com|Other Instances
-|-
image<br/>image|image<br/>image

**Pull request merged**